apps/system/js/advanced_telemetry.js

```diff
@@ -0,0 +1,426 @@
+      console.log.apply(console, args);
+    }
+  }
+
+  // This info level is more of a information level.  It's recommended to
+  // turn this on just to see general events related to initializations,
+  // and sending of payloads, etc.
+  function loginfo(...args) {
+    if(AT.LOGINFO) {
+      args.unshift('[AdvancedTelemetryInfo]');
+      console.log.apply(console, args);
+    }
+  }
+
+  // Debugging function to handle the case where the output length exceeds that
+  // which the adb console can handle.
+  function debuglongLine(longLine) {
+    if (AT.DEBUG) {
+      var array = longLine.match(/.{1,1000}/g);
+      array.forEach(function (value) {
+        console.log('[AdvancedTelemetry]' + value);
+      });
+    }
+  }
+
+  // This value is what turns advanced telemetry on/off
+  AT.TELEMETRY_ENABLED_KEY = 'debug.performance_data.advanced_telemetry';
+
+  // Base URL for sending data reports
+  // Needs to be formatted as:
+  // /submit/telemetry/id/reason/appName/appUpdateChannel/appVersion/appBuildID
+  AT.REPORT_URL = 'https://incoming.telemetry.mozilla.org/submit/telemetry';
+//  AT.REPORT_URL = 'http://10.19.2.11:4040/submit/telemetry';
+  // How often do we try to send the reports
+  // Can be overridden with metrics.advancedtelemetry.reportInterval setting.
+//  AT.REPORT_INTERVAL = 24 * 60 * 60 * 1000;  // 1 DAY
+  AT.REPORT_INTERVAL = 2 * 60 * 1000;  // 1 DAY
+
+  // If the telemetry server does not respond within this amount of time
+  // just give up and try again later.
+  // Can be overridden with metrics.advancedtelemetry.reportTimeout setting.
+  AT.REPORT_TIMEOUT = 20 * 1000;             // 20 seconds
+
+  // If sending a report fails (even though the we're online) how
+  // long do we wait before trying it again?
+  // Can be overridden with metrics.advancedtelemetry.retryTimeout setting.
+  AT.RETRY_INTERVAL = 60 * 60 * 1000;        // 1 hour
+
+  // Telemetry payload version
+  AT.TELEMETRY_VERSION = 4;
+
+  // App name (static for Telemetry)
+  AT.TELEMETRY_APP_NAME = 'FirefoxOS';
+
+
+  AT.prototype.start = function start() {
+    loginfo('Starting AdvancedTelemetry');
+    this.advancedTelemetryEnabledListener =
+      function advancedTelemetryEnabledListener(enabled) {
+        if (enabled) {
+          loginfo('Start Collecting');
+          this.startCollecting();
+        }
+        else {
+          loginfo('Stop Collecting');
+          this.stopCollecting();
+        }
+      }.bind(this);
+
+    SettingsListener.observe(AT.TELEMETRY_ENABLED_KEY,
+      false, this.advancedTelemetryEnabledListener);
+  };
+
+  AT.prototype.startCollecting = function startCollecting(done) {
+    var self = this;
+    // If we're already running there is nothing to start
+    if (this.collecting) {
+      return;
+    }
+    this.collecting = true;
+
+    this.metrics = new BatchTiming(true);
+
+    getDeviceID();
+
+    function getDeviceID() {
+      asyncStorage.getItem(DEVICE_ID_KEY, function(value) {
```

Comment:
Yes, but it seems like there may be a change as per discussions in 1171685.  I'm hoping we can just use the same implementations across ftu, aum, and advanced telemetry... So, I'll probably have a follow on for this.